src/components/AudioPlayer/AudioPlayer.container.js

```diff
@@ -1,51 +1,68 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, useState } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 
-import { setIsPlaying, playNextSong } from 'store/actions/playerActions';
+import {
+  setIsPlaying, playNextSong, setRepeat, setRandom, playPreviousSong,
+} from 'store/actions/playerActions';
 
 import AudioPlayerComponent from './AudioPlayer.component';
 
 const AudioPlayerContainer = () => {
   const songLink = useSelector((state) => state.player.get('songLink'));
+  const songsQueue = useSelector((state) => state.songsQueue.get('queue'));
+  const isPlaying = useSelector((state) => state.player.get('isPlaying'));
+  const isRandomEnabled = useSelector((state) => state.player.get('isRandomEnabled'));
+  const isRepeatEnabled = useSelector((state) => state.player.get('isRepeatEnabled'));
+  const isDisabled = songsQueue.size <= 0;
   const dispatch = useDispatch();
+  const [audio] = useState(new Audio(songLink));
 
-  const audioRef = new Audio(songLink);
+  const playerActions = {
+    forward: () => dispatch(playNextSong()),
+    rewind: () => dispatch(playPreviousSong()),
+    repeat: () => dispatch(setRepeat(!isRepeatEnabled)),
+    random: () => dispatch(setRandom(!isRandomEnabled)),
+    play: () => {
+      if (isPlaying) {
+        dispatch(setIsPlaying(false));
+      } else {
+        dispatch(setIsPlaying((true)));
+      }
+    },
+  };
 
   useEffect(() => {
-    const playMusic = () => {
-      audioRef.play()
-        .then(() => {
-          dispatch(setIsPlaying(true));
-        })
-        .catch(() => {
-          dispatch(setIsPlaying(false));
-        });
-    };
-
-    const onPause = () => {
-      if (audioRef.ended) {
-        dispatch(playNextSong());
-      }
-    };
+    audio.src = songLink;
+  }, [songLink, audio, audio.src]);
 
-    audioRef.addEventListener('canplaythrough', playMusic, false);
-    audioRef.addEventListener('pause', onPause, false);
+  useEffect(() => {
+    if (songLink) {
+      if (isPlaying) {
+        audio.play();
+      } else {
+        audio.pause();
+      }
+    }
```

Comment:
I believe the statements are flipped

Reply:
nit:

```js
if (!songLink) return;

if (isPlaying) {
  audio.pause();
} else {
  audio.play();
}
```

Reply:
Ugh yes, I had it as a ternary operation but I got linter issues when I did the refactor I did it without thinking haha